### ISSUE TITLE
Enabling proguard in d15-7 causes crash on startup

### ISSUE THREAD

**Issue by pjcollins** [xamarin/xamarin-android#1331]
Projects which enable proguard are crashing on startup when using d15-7. This is a regression as compared to d15-6.

### Steps to Reproduce

1.  msbuild /t:Install /p:Configuration=Release /p:EnableProguard=true *.csproj 

### Version Information
Xamarin.Android SDK   8.3.0.0 (HEAD/a13b88d76)

### Log File

Diagnostic installation output:
https://gist.github.com/pjcollins/c9a04e5da40e3b82379434abb07d69a0
```
02-22 14:52:56.680  1163  2477 I ActivityManager: Start proc 16357:mono.samples.button/u0a311 for activity mono.samples.button/md5e2fd3b03328de3df97c99ffd13b29e67.ButtonActivity
02-22 14:52:56.692  1859  1859 V StatusBar: setLightsOn(true)
02-22 14:52:56.728  1859  1859 V StatusBar: setLightsOn(true)
02-22 14:52:56.763 16357 16357 I chatty  : uid=10311(mono.samples.button) expire 1 line
02-22 14:52:56.794 16357 16357 W monodroid-gc: GREF GC Threshold: 46080
02-22 14:52:56.800 16357 16357 W monodroid: Calling into managed runtime init
02-22 14:52:56.829 16357 16357 W .samples.button: type=1400 audit(0.0:50169): avc: denied { read } for name="uptime" dev="proc" ino=4026532080 scontext=u:r:untrusted_app_25:s0:c512,c768 tcontext=u:object_r:proc:s0 tclass=file permissive=0
02-22 14:52:56.938 16357 16357 E mono    :
02-22 14:52:56.938 16357 16357 E mono    : Unhandled Exception:
02-22 14:52:56.938 16357 16357 E mono    : System.TypeInitializationException: The type initializer for 'Java.Interop.ManagedPeer' threw an exception. ---> System.ArgumentException: Handle must be valid.
02-22 14:52:56.938 16357 16357 E mono    : Parameter name: type
02-22 14:52:56.938 16357 16357 E mono    :   at Java.Interop.JniEnvironment+Types.IsInstanceOf (Java.Interop.JniObjectReference instance, Java.Interop.JniObjectReference type) [0x00022] in <e5da7a7a83124ab3827f81a186993213>:0
02-22 14:52:56.938 16357 16357 E mono    :   at Android.Runtime.JNIEnv.IsInstanceOf (System.IntPtr obj, System.IntPtr clazz) [0x0000e] in <546fec40177a4027a686650d5da2a824>:0
02-22 14:52:56.938 16357 16357 E mono    :   at Java.Interop.Runtime.IsGCUserPeer (System.IntPtr value) [0x0000f] in <546fec40177a4027a686650d5da2a824>:0
02-22 14:52:56.938 16357 16357 E mono    :   at Java.Interop.Runtime.IsGCUserPeer (Android.Runtime.IJavaObject value) [0x0000b] in <546fec40177a4027a686650d5da2a824>:0
02-22 14:52:56.938 16357 16357 E mono    :   at Java.Interop.TypeManager.CreateInstance (System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer, System.Type targetType) [0x00127] in <546fec40177a4027a686650d5da2a824>:0
02-22 14:52:56.938 16357 16357 E mono    :   at Java.Lang.Object.GetObject (System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer, System.Type type) [0x000b9] in <546fec40177a4027a686650d5da2a824>:0
02-22 14:52:56.938 16357 16357 E mono    :   at Java.Lang.Object._GetObject[T] (System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer) [0x00017] in <546fec40177a4027a686650d5da2a824>:0
02-22 14:52:56.938 16357 16357 E mono    :   at Java.Lang.Object.GetObject[T] (System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer) [0x00000] in <546fec40177a4027a686650d5da2a824>:0
02-22 14:52:56.938 16357 16357 E mono    :   at Android.Runtime.AndroidRuntime.GetExceptionForThrowable (Java.Interop.JniObjectReference& value, Java.Interop.JniObjectReferenceOptions transfer) [0x00006] in <546fec40177a4027a686650d5da2a824>:0
02-22 14:52:56.938 16357 16357 E mono    :   at Java.Interop.JniEnvironment+Types.FindClass (System.String classname) [0x0010c] in <e5da7a7a83124ab3827f81a186993213>:0
02-22 14:52:56.938 16357 16357 E mono    :   at Java.Interop.JniType..ctor (System.String classname) [0x00006] in <e5da7a7a83124ab3827f81a186993213>:0
02-22 14:52:56.938 16357 16357 E mono    :   at Java.Interop.JniType.GetCachedJniType (Java.Interop.JniType& cachedType, System.String classname) [0x00018] in <e5da7a7a83124ab3827f81a186993213>:0
02-22 14:52:56.938 16357 16357 E mono    :   at Java.Interop.JniPeerMembers.get_JniPeerType () [0x0000c] in <e5da7a7a83124ab3827f81a186993213>:0
02-22 14:52:56.938 16357 16357 E mono    :   at Java.Interop.ManagedPeer..cctor () [0x00019] in <e5da7a7a83124ab3827f81a186993213>:0
02-22 14:52:56.938 16357 16357 E mono    :    --- End of inner exception stack trace ---
02-22 14:52:56.938 16357 16357 E mono    :   at Java.Interop.JniRuntime..ctor (Java.Interop.JniRuntime+CreationOptions options) [0x00229] in <e5da7a7a83124ab3827f81a186993213>:0
02-22 14:52:56.938 16357 16357 E mono    :   at Android.Runtime.AndroidRuntime..ctor (System.IntPtr jnienv, System.IntPtr vm, System.Boolean allocNewObjectSupported, System.IntPtr classLoader, System.IntPtr classLoader_loadClass) [0x0000d] in <546fec40177a4027a686650d5da2a824>:0
02-22 14:52:56.938 16357 16357 E mono    :   at Android.Runtime.JNIEnv.Initialize (Android.Runtime.JnienvInitializeArgs* args) [0x0017c] in <546fec40177a4027a686650d5da2a824>:0
02-22 14:52:56.939 16357 16357 E mono-rt : [ERROR] FATAL UNHANDLED EXCEPTION: System.TypeInitializationException: The type initializer for 'Java.Interop.ManagedPeer' threw an exception. ---> System.ArgumentException: Handle must be valid.
02-22 14:52:56.939 16357 16357 E mono-rt : Parameter name: type
02-22 14:52:56.939 16357 16357 E mono-rt :   at Java.Interop.JniEnvironment+Types.IsInstanceOf (Java.Interop.JniObjectReference instance, Java.Interop.JniObjectReference type) [0x00022] in <e5da7a7a83124ab3827f81a186993213>:0
02-22 14:52:56.939 16357 16357 E mono-rt :   at Android.Runtime.JNIEnv.IsInstanceOf (System.IntPtr obj, System.IntPtr clazz) [0x0000e] in <546fec40177a4027a686650d5da2a824>:0
02-22 14:52:56.939 16357 16357 E mono-rt :   at Java.Interop.Runtime.IsGCUserPeer (System.IntPtr value) [0x0000f] in <546fec40177a4027a686650d5da2a824>:0
02-22 14:52:56.939 16357 16357 E mono-rt :   at Java.Interop.Runtime.IsGCUserPeer (Android.Runtime.IJavaObject value) [0x0000b] in <546fec40177a4027a686650d5da2a824>:0
02-22 14:52:56.939 16357 16357 E mono-rt :   at Java.Interop.TypeManager.CreateInstance (System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer, System.Type targetType) [0x00127] in <546fec40177a4027a686650d5da2a824>:0
02-22 14:52:56.939 16357 16357 E mono-rt :   at Java.Lang.Object.GetObject (System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer, System.Type type) [0x000b9] in <546fec40177a4027a686650d5da2a824>:0
02-22 14:52:56.939 16357 16357 E mono-rt :   at Java.Lang.Object._GetObject[T] (System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer) [0x00017] in <546fec40177a4027a686650d5da2a824>:0
02-22 14:52:56.939 16357 16357 E mono-rt :   at Java.Lang.Object.GetObject[T] (System.IntPtr handle, Android.Runtime.JniHandleOwnership transfer) [0x00000] in <546fec40177a4027a686650d5da2a824>:0
02-22 14:52:56.939 16357 16357 E mono-rt :   at Android.Runtime.AndroidRuntime.GetExceptionForThrowable (Java.Interop.JniObjectReference& value, Java.Interop.JniObjectReferenceOptions transfer) [0x00006] in <546fec40177a4027a686650d5da2a824>:0
02-22 14:52:56.939 16357 16357 E mono-rt :   at Java.Interop.JniEnvironment+Types.FindClass (System.String classname) [0x0010c] in <e5da7a7a83124ab3827f81a186993213>:0
02-22 14:52:56.939 16357 16357 E mono-rt :   at Java.Interop.JniType..ctor (System.String classname) [0x00006] in <e5da7a7a83124ab3827f81a186993213>:0
02-22 14:52:56.939 16357 16357 E mono-rt :   at Java.Interop.JniType.GetCachedJniType (Java.Interop.JniType& cachedType, System.String classname) [0x00018] in <e5da7a7a83124ab3827f81a186993213>:0
02-22 14:52:56.939 16357 16357 E mono-rt :   at Java.Interop.JniPeerMembers.get_JniPeerType () [0x0000c] in <e5da7a7a83124ab3827f81a186993213>:0
02-22 14:52:56.939 16357 16357 E mono-rt :   at Java.Interop.ManagedPeer..cctor () [0x00019] in <e5da7a7a83124ab3827f81a186993213>:0
02-22 14:52:56.939 16357 16357 E mono-rt :    --- End of inner exception stack trace ---
02-22 14:52:56.939 16357 16357 E mono-rt :   at Java.Interop.JniRuntime..ctor (Java.Interop.JniRuntime+CreationOptions options) [0x00229] in <e5da7a7a83124ab3827f81a186993213>:0
02-22 14:52:56.939 16357 16357 E mono-rt :   at Android.Runtime.AndroidRuntime..ctor (System.IntPtr jnienv, System.IntPtr vm, System.Boolean allocNewObjectSupported, System.IntPtr classLoader, System.IntPtr classLoader_loadClass) [0x0000d] in <546fec40177a4027a686650d5da2a824>:0
02-22 14:52:56.939 16357 16357 E mono-rt :   at Android.Runtime.JNIEnv.Initialize (Android.Runtime.JnienvInitializeArgs* args) [0x0017c] in <546fec40177a4027a686650d5da2a824>:0
02-22 14:52:56.972  1163  2477 I ActivityManager: Process mono.samples.button (pid 16357) has died: fore TOP
02-22 14:52:56.972   762   762 I Zygote  : Process 16357 exited cleanly (1)
02-22 14:52:56.973  1163  1179 W zygote64: kill(-16357, 9) failed: No such process
```

**Comment by radekdoulik** [comment]
Looks like we need to tell proguard to keep ManagedPeer class around.

**Comment by radekdoulik** [comment]
I will add it to the https://github.com/xamarin/xamarin-android/blob/master/src/Xamarin.Android.Build.Tasks/Resources/proguard_xamarin.cfg

**Comment by jonpryor** [comment]
This fix has been cherry-picked to the d15-7 branch as 544cb6df9cdb0c1d3104df8de18f3b961f9f15b0.

**Comment by pjcollins** [comment]
Verified fixed in XA 8.3.0.10.
